[BUG] [Reproduction & Videos Provided] Dragging file to upload dropzone when drawer is closed throws error

### Describe the Bug

**Description:**
When dragging a file into the upload dropzone while the drawer is not opened, an error occurs:

**Cannot read properties of undefined**

Please refer to Video 1 – this issue only happens if the file is dropped directly onto the upload dropzone while the drawer is still closed.

In contrast, as shown in Video 2, if you click "Create New" first and open the drawer, then drag and drop a file into the upload dropzone, the file upload works correctly and no error is thrown.

Version Info:
Affected version: v3.58.0
Last working version: v3.54.0

### Link to the code that reproduces this issue

https://github.com/mydansun/payload-drag-error

### Reproduction Steps

## Video 1:

1. Navigate to the new **Post** page with the upload dropzone.
2. **DO NOT** click create new
3. Drag and drop a file onto the upload dropzone.
4. Observe the error.

https://github.com/user-attachments/assets/fd1d9511-a56f-4e6f-80b8-e6a759c8121c

## Video 2:

if you **click "Create New" first** and open the drawer, then drag and drop a file into the upload dropzone, the file upload works correctly and no error is thrown.

https://github.com/user-attachments/assets/cc846bcb-b7de-4ac3-8d32-8d9d0802a5a6

### Which area(s) are affected? (Select all that apply)

area: ui

### Environment Info

```text
Unable to run `pnpm payload info` due to `ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command "payload" not found`
```

--- a/packages/ui/src/fields/Upload/RelationshipContent/index.tsx
+++ b/packages/ui/src/fields/Upload/RelationshipContent/index.tsx
@@ -89,7 +89,7 @@ export function RelationshipContent(props: Props) {
   }
 
   const metaText = withMeta ? generateMetaText(mimeType, byteSize) : ''
-  const previewAllowed = displayPreview ?? collectionConfig.upload?.displayPreview ?? true
+  const previewAllowed = displayPreview ?? collectionConfig?.upload?.displayPreview ?? true
 
   return (
     <div className={[baseClass, className].filter(Boolean).join(' ')}>

